From: Andre McCurdy <armccurdy at gmail.com>
Date: Mon, 13 Jun 2016 13:32:44 -0700
Subject: fix client->gfx_state initialisation

Shortly before the DirectFB 1.7.7 release, an optimisation was added
to CoreGraphicsStateClient_Init() to avoid creating an extended
Graphics State object if it will not later be required:

  4d422fb Client: Create extended Graphics State object when needed for later usage

Unfortunately the client->gfx_state variable used to track the
extended Graphics State object is not initialised, which can lead to
rashes etc due to creation of the Graphics State object erroneously
being skipped.

Upstream-Status: Pending

Signed-off-by: Andre McCurdy <armccurdy at gmail.com>
---
 src/core/CoreGraphicsStateClient.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/core/CoreGraphicsStateClient.cpp b/src/core/CoreGraphicsStateClient.cpp
index 5d46f0e..5007755 100644
--- a/src/core/CoreGraphicsStateClient.cpp
+++ b/src/core/CoreGraphicsStateClient.cpp
@@ -364,6 +364,7 @@ CoreGraphicsStateClient_Init( CoreGraphicsStateClient *client,
      client->renderer  = NULL;
      client->requestor = NULL;
      client->throttle  = NULL;
+     client->gfx_state = NULL;
 
      if (dfb_config->task_manager) {
           if (dfb_config->call_nodirect) {
